tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/RunCspellTool/RunCspellTool.cs

```diff
@@ -0,0 +1,290 @@
+            description: "The path to the package to check for spelling"
+        )
+        {
+            Arity = ArgumentArity.ExactlyOne
+        };
+
+        public RunCspellTool(ILogger<RunCspellTool> logger, IOutputService output) : base()
+        {
+            this.logger = logger;
+            this.output = output;
+            
+            CommandHierarchy = [SharedCommandGroups.Checks];
+        }
+
+        public override Command GetCommand()
+        {
+            Command command = new("cspell", "Run cspell to check for typos in the specified package");
+            command.AddArgument(_packagePathArg);
+            command.SetHandler(async ctx => { await HandleCommand(ctx, ctx.GetCancellationToken()); });
+            return command;
+        }
+
+        public override async Task HandleCommand(InvocationContext ctx, CancellationToken ct)
+        {
+            string packagePath = ctx.ParseResult.GetValueForArgument(_packagePathArg);
+            var result = await RunCspellCheck(packagePath);
+            ctx.ExitCode = ExitCode;
+            output.Output(result);
+        }
+
+        [McpServerTool(Name = "run-cspell"), Description("Run cspell to check for typos in the specified package path")]
+        public async Task<DefaultCommandResponse> RunCspellCheck(string packagePath)
+        {
+            try
+            {
+                logger.LogInformation("Running cspell check for package path: {packagePath}", packagePath);
+
+                if (string.IsNullOrWhiteSpace(packagePath))
+                {
+                    SetFailure(1);
+                    return new DefaultCommandResponse
+                    {
+                        ResponseError = "Package path is required"
+                    };
+                }
+
+                if (!Directory.Exists(packagePath))
+                {
+                    SetFailure(1);
+                    return new DefaultCommandResponse
+                    {
+                        ResponseError = $"Package path does not exist: {packagePath}"
+                    };
+                }
+
+                // Get the absolute path to the Invoke-Cspell.ps1 script
+                var scriptPath = GetInvokeCspellScriptPath();
+                if (string.IsNullOrEmpty(scriptPath))
+                {
+                    SetFailure(1);
+                    return new DefaultCommandResponse
+                    {
+                        ResponseError = "Could not locate Invoke-Cspell.ps1 script"
+                    };
+                }
+
+                var startTime = DateTime.UtcNow;
+                var output = await RunPowerShellScript(scriptPath, packagePath);
+                var duration = (DateTime.UtcNow - startTime).TotalMilliseconds;
+
+                // Parse the cspell output to determine success/failure and extract typos
+                var result = ParseCspellOutput(output);
+
+                return new DefaultCommandResponse
```

Comment:
Should we call `SetFailure` if `result.Success` is false so we get a non-zero exit code?